src/family_name_classifier.py

```diff
@@ -0,0 +1,315 @@
+import builtins
+
+import pandas as pd
+import numpy as np
+import itertools
+import unicodedata
+import re
+import collections
+import random
+import sklearn.feature_extraction
+import sklearn.linear_model
+import sklearn.cross_validation
+import sklearn.externals.joblib
+random.seed(0)
+
+
+class FamilyNameClassifier:
+    IGNORE_LIST = {'de', 'da', 'do', 'des', 'das', 'dos', 'e', 'dr', 'sobrinho', 'junior', 'neto', 'jr', 'filho'}
+    NOT_ACCEPTABLE_CHARACTERS = re.compile(r'[^a-z ]')
+
+    def __init__(self):
+        self.vectorizer = None
+        self.surname_counter = None
+        self.total_surnames = None
+        self.total_first_names = None
+        self.model = None
+
+    def tokenize(self, name):
+        """"
+        Simple tokenizer that normalizes and split a name by spaces.
+
+        :param name: string
+        :return: tokens: list
+        """
+
+        name = name.strip().lower()
+        name = unicodedata.normalize('NFD', name)
+        name = self.NOT_ACCEPTABLE_CHARACTERS.sub("", name)
+        return [n for n in name.split(' ') if n not in self.IGNORE_LIST and len(n) > 1]
+
+    def __count_names(self, relatives_dict):
+        """
+        Calculate some names statistics used internally.
+
+        :param relatives_dict: dict
```

Comment:
What's the _shape_ of this dict? `{'congressperson name': ["congressperson's mon", "congressperson's dad"]}`?